src/main/scala/nl/knaw/dans/bag/DansBag.scala

```diff
@@ -422,12 +490,33 @@ trait DansBag {
    * Please note that, while the new file is added to the bag immediately, the changes to the
    * payload manifests will only be applied to the bag on the file system once [[DansBag#save]] is called.
    *
+   * @example
+   * {{{ dir.children.map(f => bag.addPayloadFile(f, dir.relativize(f))) }}}
```

Comment:
As I saw a recursion, so I suppose this can be simplified to `bag.addPayloadFile(dir, ???)` See also [deposit-api](https://github.com/DANS-KNAW/easy-deposit-api/blob/0e2bea9d7685fed0874d28e3a2b4e294c75e533f/src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala#L77)

Reply:
you are completely correct! I'll make sure to add that to the documentation and the examples as well!

By the way, we already had a test for that.

https://github.com/DANS-KNAW/dans-bag-lib/blob/00ea4f0af90d5d5785bac99cdceeec4ac087969a/src/test/scala/nl/knaw/dans/bag/v0/DansV0BagSpec.scala#L1695-L1720

Reply:
same is true, btw, for `bag.addTagFile(dir, ???)`